lib/blocktree/blocktree.go

```diff
@@ -133,6 +100,39 @@ func (bt *BlockTree) AddBlock(block *types.Block, arrivalTime uint64) error {
 	return nil
 }
 
+// GetAllBlocksAtDepth will return all blocks hashes with the same depth as the given block hash
+func (bt *BlockTree) GetAllBlocksAtDepth(hash common.Hash) []common.Hash {
+	hashes := []common.Hash{}
+
+	depth := bt.getNode(hash).depth
+
+	if bt.head.depth.Cmp(depth) == 0 {
+		hashes = append(hashes, bt.head.hash)
+		return hashes
+	}
+
+	return bt.head.getNodesWithDepth(depth, hashes)
+}
+
+// getNodesWithDepth returns all descendent nodes with the desired depth
+func (n *node) getNodesWithDepth(depth *big.Int, hashes []common.Hash) []common.Hash {
```

Comment:
are you sure that a func for  (n *node) should be together with others for (bt *BlockTree) in the same file ?

Reply:
it's been moved

Reply:
oh cool